tracer/src/Datadog.Trace/Telemetry/TelemetryTransportFactory.cs

```diff
@@ -22,13 +22,14 @@ public TelemetryTransportFactory(Uri baseEndpoint, string apiKey)
 
         public ITelemetryTransport Create()
         {
+            var timeout = TimeSpan.FromSeconds(10);
```

Comment:
nit - I would have used 15 seconds everywhere so that we don't have to think about the specific cases and as 10 seconds is already a lot.

Reply:
> as 10 seconds is already a lot.

How do you feel about 5s?

Reply:
same comment :) I'd rather the same parameter (thus 15 sec) everywhere unless there's a clear reason why we shouldn't. 
Also have you asked Perf & Reliability or the Event Platform intake team if they had any reco even though I clearly don't think they would say 15 secs

Reply:
>I'd rather the same parameter (thus 15 sec) everywhere unless there's a clear reason why we shouldn't.

The "clear reason" for a shorter timeout for telemetry IMO is that we send telemetry on app-close, so any timeout here will be blocking app shutdown. Also, it's not as bad to "lose" telemetry requests. 

> Also have you asked Perf & Reliability or the Event Platform intake team if they had any reco even though I clearly don't think they would say 15 secs.

No 😳  But CIApp (cc @tonyredondo) have IIRC, and that's where the 15s came from there. But with telemetry I feel like shorter would be better. 

As an aside, I'm thinking a separate PR in which we explicitly use a shorter timeout (and no retries) on app close would make sense.

Reply:
Ok. You've convinced me :)

Reply:
Went with your original suggestion in https://github.com/DataDog/dd-trace-dotnet/commit/c9eb514ec905a16b8edbd546c7ddeff094c2070e and updated to be 15s everywhere for now.